api/grpcserver/transaction_service.go

```diff
@@ -41,6 +43,14 @@ func NewTransactionService(
 func (s TransactionService) SubmitTransaction(ctx context.Context, in *pb.SubmitTransactionRequest) (*pb.SubmitTransactionResponse, error) {
 	log.Info("GRPC TransactionService.SubmitTransaction")
 
+	if len(in.Transaction) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "`Transaction` payload is empty")
+	}
+
+	if s.Mesh.ProcessedLayer() != s.Mesh.LatestLayerInState() {
```

Comment:
For the right way to check if the node is synced, take a look here:

https://github.com/spacemeshos/go-spacemesh/blob/a1f005549ddf99ceaf24724d93a306930db1460d/api/grpcserver/node_service.go#L121

Reply:
Thanks I will definitely  use the syncer impl to check if the node is synced. Looks like the `IsSynced` does somewhat the same check so at least I was on the right track (almost).